integrations/cohere/src/haystack_integrations/components/generators/cohere/chat/chat_generator.py

```diff
@@ -359,21 +344,29 @@ async def _parse_async_streaming_response(
     """
     Parses Cohere's async streaming chat response into a Haystack ChatMessage.
     """
-    state = _initialize_streaming_state()
+    chunks: List[StreamingChunk] = []
+    tool_call_index = -1
 
     async for chunk in response:
-        stream_content = _process_cohere_chunk(cohere_chunk=chunk, state=state, model=model)
-        if stream_content:
-            stream_chunk = StreamingChunk(content=stream_content, component_info=component_info)
-            await streaming_callback(stream_chunk)
+        if chunk.type == "tool-call-start":
+            tool_call_index += 1
+
+        streaming_chunk = _convert_cohere_chunk_to_streaming_chunk(
+            chunk=chunk, component_info=component_info, model=model
+        )
```

Comment:
We need to pass tool_call_index here.

```suggestion
        streaming_chunk = _convert_cohere_chunk_to_streaming_chunk(
            chunk=chunk, component_info=component_info, model=model,
            tool_call_index=tool_call_index
        )
```